Update foreach max/min

izdeby opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #52450 wip
* #49039 [wip] Replace optimizers in torch.optim with the ones from torch.optim._multi_tensor
* #52449 [WIP] Add type promotion to a fast path for foreach APIs
* #52448 [WIP] Support broadcasting via slow path
* #48224 [wip] Pushed unsupported scenarios to slow path
* #51170 Updated foreach pointwise ops to support complex and bloat16 on CUDA
* **#49714 Update foreach max/min**
* #51059 Refactor foreach binary ops tests with tensor lists to use OpInfo
* #51058 Refactor foreach binary ops tests with scalars to use OpInfo

Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min
- Updated the tests to use OpInfo

Commit:
Update foreach max/min

[ghstack-poisoned]

zou3519 commented on test/test_foreach.py:
```diff
@@ -1054,25 +1054,32 @@ def test_addcdiv(self, device, dtype):
         self._test_pointwise_op(device, dtype, torch._foreach_addcdiv, torch._foreach_addcdiv_, torch.addcdiv)
 
-    @dtypes(*torch.testing.get_all_dtypes(include_bfloat16=False, include_bool=False, include_complex=False))
+    @dtypes(*torch.testing.get_all_dtypes())
```
What added bfloat16 and bool support? Were those supported all along?

izdeby replied:
I've added support for torch.bool and torch.bfloat16 in this PR

zou3519 commented on test/test_foreach.py:
```diff
@@ -1054,25 +1054,32 @@ def test_addcdiv(self, device, dtype):
         self._test_pointwise_op(device, dtype, torch._foreach_addcdiv, torch._foreach_addcdiv_, torch.addcdiv)
 
-    @dtypes(*torch.testing.get_all_dtypes(include_bfloat16=False, include_bool=False, include_complex=False))
+    @dtypes(*torch.testing.get_all_dtypes())
```
Is there a type promotion test for foreach_maximum / foreach_minimum, or is that out of the scope of this PR?

izdeby replied:
Type promotion will be tested in upcoming PRs once we push different dtype behavior from error to slow path.

izdeby committed:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

[ghstack-poisoned]

Commit:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

[ghstack-poisoned]

Commit:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

[ghstack-poisoned]

Commit:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

[ghstack-poisoned]

Commit:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

[ghstack-poisoned]

Commit:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min

[ghstack-poisoned]

Commit:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min

[ghstack-poisoned]

Commit:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min

[ghstack-poisoned]

Commit:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min

[ghstack-poisoned]

Commit:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min

[ghstack-poisoned]

Commit:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min

[ghstack-poisoned]

Commit:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min
- Updated the tests to use OpInfo

[ghstack-poisoned]

izdeby committed:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min
- Updated the tests to use OpInfo

[ghstack-poisoned]

Commit:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min
- Updated the tests to use OpInfo

[ghstack-poisoned]

Commit:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min
- Updated the tests to use OpInfo

[ghstack-poisoned]

Commit:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min
- Updated the tests to use OpInfo

[ghstack-poisoned]

Commit:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min
- Updated the tests to use OpInfo

[ghstack-poisoned]

Commit:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min
- Updated the tests to use OpInfo

[ghstack-poisoned]

Commit:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min
- Updated the tests to use OpInfo

[ghstack-poisoned]

Commit:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min
- Updated the tests to use OpInfo

[ghstack-poisoned]

Commit:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min
- Updated the tests to use OpInfo

[ghstack-poisoned]

Commit:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min
- Updated the tests to use OpInfo

[ghstack-poisoned]

izdeby committed:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min
- Updated the tests to use OpInfo

[ghstack-poisoned]

izdeby committed:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min
- Updated the tests to use OpInfo

[ghstack-poisoned]

izdeby committed:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min
- Updated the tests to use OpInfo

[ghstack-poisoned]

izdeby committed:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min
- Updated the tests to use OpInfo

[ghstack-poisoned]

izdeby committed:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min
- Updated the tests to use OpInfo

[ghstack-poisoned]

izdeby committed:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min
- Updated the tests to use OpInfo

[ghstack-poisoned]

izdeby committed:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min
- Updated the tests to use OpInfo

[ghstack-poisoned]

izdeby committed:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min
- Updated the tests to use OpInfo

[ghstack-poisoned]

Commit:
Update on "Update foreach max/min"


Differential Revision: [D25674139](https://our.internmc.facebook.com/intern/diff/D25674139)

-------
- Add support for torch.bool and torch.bfloat16 for foreach max/min
- Updated the tests to use OpInfo

[ghstack-poisoned]
